src/test/test_step.go

```diff
@@ -504,7 +504,10 @@ func parseTestOutput(stdout string, stderr string, runError error, duration time
 }
 
 // Parses the coverage output for a single target.
-func parseCoverageFile(target *core.BuildTarget, coverageFile string, run int) *core.TestCoverage {
+func parseCoverageFile(state *core.BuildState, target *core.BuildTarget, coverageFile string, run int) *core.TestCoverage {
+	if !target.NeedCoverage(state) {
+		return core.NewTestCoverage()
```

Comment:
Why not return nil?

Reply:
I don't think the caller is expecting to get nil; this matches what happens in the error case below.

Reply:
I'd expect anything that's returning a pointer and an error to always return nil if it returns an error, but yes, having looked at `parseTestCoverageFile` that's not the case here